### PR TITLE
When swapping out ShellItems clear services

### PR DESCRIPTION
**Pull request by PureWeen**
### Description

When the user is switching to a new context on shell and they've loaded the ShellContent via the services out. This will clear out the ContentCache so the service will get retrieved again. This allows the users to specify if they want pages to be transient vs persistent. 

#### Rules for clearing out the cache
- If the window changes for a shell  (this will happen when the user is swapping out the shell)
- If the user switches to a new ShellItem (new TabBar, new FlyoutItem)
  -  The cache will not clear out when switch tabs. It only clears the cache when the app switches to a new "ShellItem" level context
- If the user sets the `ShellItem => ShellSection => ShellContent` to not be visible then we'll clear out the cache because the user has indicated that this Page should not longer be part of the active structure

#### Scenarios where the cache isn't cleared
- Navigating through bottom tabs
- Navigation through top tabs
- if the user hasn't registered the Page itself against the services then we don't clear out the cache. This means for anyone not registering shell services, the behavior won't change.

### CODE REVIEW

**Comment by albyrock87** [comment]
What about adding the `IServiceCollection` to itself as a reference so that later with the `IServiceProvider` we can see how a given type was registered? (Scoped/Transient)

File: src/Controls/src/Core/Shell/ShellContent.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Collections.Specialized;
 using System.ComponentModel;
 using System.Reflection;
+using System.Security.Cryptography;
```
Comment (Redth) [inline src/Controls/src/Core/Shell/ShellContent.cs:8]: Is this used anywhere?

Comment (PureWeen) [reply src/Controls/src/Core/Shell/ShellContent.cs]: Removed, VSCode sometimes likes to add stuff :-(

**Review by rmarinho — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Shell/ShellContent.cs [hunk 2]
```diff
@@ -187,18 +199,79 @@ Page ContentCache
...
+			if (_contentCache is not null)
+			{
+				_contentCache.Unloaded -= OnPageUnloaded;
+				RemoveLogicalChild(_contentCache);
+			}
+
+			_contentCache = null;
+		}
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
```
[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/src/Core/Shell/ShellContent.cs]: fixed

**Review by rmarinho — APPROVED** [review]

**Pull request merged**